src/main/java/pro/trevor/tankgame/rule/impl/Rules.java

```diff
@@ -0,0 +1,233 @@
+package pro.trevor.tankgame.rule.impl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import pro.trevor.tankgame.rule.definition.ConditionalRule;
+import pro.trevor.tankgame.rule.definition.MetaTickActionRule;
+import pro.trevor.tankgame.rule.definition.TickActionRule;
+import pro.trevor.tankgame.rule.definition.player.PlayerActionRule;
+import pro.trevor.tankgame.state.board.Board;
+import pro.trevor.tankgame.state.board.Position;
+import pro.trevor.tankgame.state.board.floor.AbstractPositionedFloor;
+import pro.trevor.tankgame.state.board.floor.GoldMine;
+import pro.trevor.tankgame.state.board.unit.EmptyUnit;
+import pro.trevor.tankgame.state.board.unit.Tank;
+import pro.trevor.tankgame.state.board.unit.Wall;
+import pro.trevor.tankgame.state.meta.Council;
+import pro.trevor.tankgame.state.range.DonateTankRange;
+import pro.trevor.tankgame.state.range.MovePositionRange;
+import pro.trevor.tankgame.state.range.TankRange;
+import pro.trevor.tankgame.util.range.DiscreteIntegerRange;
+import pro.trevor.tankgame.util.range.IntegerRange;
+
+import static pro.trevor.tankgame.util.Util.*;
+
+public class Rules 
+{
+    public static class Conditional 
+    {
+        public static ConditionalRule<Wall> DESTROY_WALL_ON_ZERO_DURABILITY = new ConditionalRule<>((s, w) -> w.getDurability() == 0, 
```

Comment:
General question: can these be declared final? They should be if they can.

Reply:
Yeah, should be possible.